src/components/Typography/Typography.tsx

```diff
@@ -0,0 +1,65 @@
+import {css} from '@emotion/react';
+import {mq} from '@theme/index';
+import {PropsWithChildren} from 'react';
+import {TypographyProps} from './TypographyProps';
+
+export const styles = {
+    h1: css({
+        fontSize: '3rem',
+        lineHeight: '4.5rem',
+        fontWeight: '700',
+        [mq['SMALLDESKTOP']]: {
+            fontSize: '4.8rem',
+            lineHeight: '6.24rem',
+        },
+    }),
+
+    h2: css({
+        fontSize: '2rem',
+        lineHeight: '3rem',
+        fontWeight: '600',
+    }),
+
+    h3: css({
+        fontSize: '1.6rem',
+        lineHeight: '2.4rem',
+        fontWeight: '600',
+    }),
+
+    h4: css({
+        fontSize: '1.2rem',
+        lineHeight: '1.8rem',
+        fontWeight: '400',
+    }),
+
+    p: css({
+        fontSize: '1.6rem',
+        lineHeight: '2.4rem',
+        fontWeight: '400',
+    }),
+
+    th: css({
+        fontSize: '1.2rem',
+        lineHeight: '1.8rem',
+        fontWeight: '600',
+    }),
+
+    td: css({
+        fontSize: '1.2rem',
+        lineHeight: '1.8rem',
+        fontWeight: '400',
+
+        [mq['SMALLDESKTOP']]: {
+            fontSize: '1.4rem',
+            lineHeight: '2.1rem',
+        },
+    }),
+};
+
+const Typography = ({component, children, overridecss}: PropsWithChildren<TypographyProps>) => {
+    const Comp = component ?? 'p';
+
+    return <Comp css={[styles[Comp], overridecss]}>{children}</Comp>;
+};
+
```

Comment:
I read more into this: https://emotion.sh/docs/css-prop#style-precedence
Overriding it using the overrideCss prop is not the intended behavior by them, but a hacky way.
```suggestion

const Typography = ({component, children, className}: PropsWithChildren<TypographyProps>) => {
    const Comp = component ?? 'p';

    return <Comp css={styles[Comp]} className={className}>{children}</Comp>;
};

```
And pass in the CSS prop only on the Typography component

Reply:
didn't understand